File: 2.5 Programowanie Obiektowe w C++/PO 23.03.10/src/Tosoba.cpp
#include "Tosoba.h"
#include <iostream>
#include <string.h>
#include <cstdlib>
#include <cstring>

using namespace std;

Tosoba::Tosoba()
{
    cout<<"konstruktor domyslny klasy Tosoba"<<endl;
    strcpy(nazwisko, "Klimiuk");
    imie = "Bartosz";
    dataUr.d = 7;
    dataUr.m = 6;
    dataUr.r = 2003;
}

Tosoba::Tosoba(int dd, int mm, int rr)
{
    dataUr.d = dd;
    dataUr.m = mm;
    dataUr.r = rr;
}

void Tosoba::wczytaj()
{
    cout<<"imie:  ";
    cin>>imie;
    cout<<"nazwisko:  ";
    cin>>nazwisko;
    cout<<"data urodzenia:"<<endl;
    cin>>dataUr.d>>dataUr.m>>dataUr.r;
}

void Tosoba::wyswietl()
{
    cout<<endl<<imie<<endl<<nazwisko<<endl;
    cout<<dataUr.d<<"."<<dataUr.m<<"."<<dataUr.r<<" r."<<endl;
}

void Tosoba::info()
{
    int wiek = 2023 - dataUr.r;

    if (wiek > 50)
        cout<<"Osoba 50+"<<endl;
    else if (wiek > 30)
        cout<<"Osoba 30+"<<endl;
    else if (wiek >= 18)
        cout<<"Osoba dorosla"<<endl;
    else
        cout<<"Dziecko"<<endl;
    cout<<endl;
}

void Tosoba::wyswietl(int rok)
{
    cout<<endl<<imie<<endl<<nazwisko<<endl;
    cout<<rok - dataUr.r<<" lat"<<endl;
}

void Tosoba::info(int rok)
{
    cout<<"Osiagniecie pelnoletnosci w "<<dataUr.r + 18<<" roku"<<endl<<endl;
}

Tosoba::~Tosoba()
{
    cout<<"destruktor klasy Tosoba"<<endl;
}

File: 5.SE.4 Programowanie w Języku SWIFT/PWJS_24-12-13/PWJS_24-12-13/tOsoba.swift
//
//  tOsoba.swift
//  PWJS_24-12-13
//
//  Created by student on 13/12/2024.
//

class Osoba {
    private var Imie: String
    private var Nazwisko: String
    private var rokUr: UInt
    
    init(Imie: String, Nazwisko: String, rok: UInt) {
        self.Imie = Imie
        self.Nazwisko = Nazwisko
        if(rok>2024) {
            self.rokUr = 2024
        } else {
            self.rokUr = rok
        }
    }
    
    func info() {
        print("Nazwisko: \(Nazwisko)")
        print("Imie: \(Imie)")
        print("Rok urodzenia: \(rokUr)")
    }
    
    func getRokUrodzenia() -> UInt {
        return self.rokUr
    }
}
